docs/src/get-started/concepts.rst

```diff
@@ -0,0 +1,239 @@
+    :width: 550px
+    :align: center
+
+    Illustration of gradients stored inside a :py:class:`TensorBlock`.
+
+
+.. TODO: explain how the gradient sample works in a separate tutorial
+
+Components
+----------
+
+There is one more thing :py:class:`TensorBlock` can contain. When working with
+vectorial data, we also handle vector **components** in both data and metadata.
+In its most generic form, a :py:class:`TensorBlock` contains a
+:math:`N`-dimensional data array (with :math:`N \geqslant 2`), and :math:`N` set
+of :py:class:`Labels`. The first Labels describe the *samples*, the last Labels
+describe the *properties*, and all the remaining Labels describe vectorial
+**components** (matching all remaining axes, from the second to
+one-before-last).
+
+For example, gradients with respect to positions are actually a bit more complex
+than the illustration above. They always contain a supplementary axis in the
+data for the :math:`x/y/z` direction of the gradient, associated with a
+**component** :py:class:`Labels`. Getting back to the example where we store
+energy in the :py:class:`TensorBlock` values, the gradient (i.e. the negative of
+the forces) samples describe with respect to which atom position we are taking
+gradient, and the component :py:class:`Labels` allow to find the :math:`x/y/z`
+component of the forces.
+
+.. figure:: ../../static/images/TensorBlock-Components.*
+    :width: 400px
+    :align: center
+
+    Illustration of a :py:class:`TensorBlock` containing **components** as an
+    extra set of metadata to describe supplementary axes of the data array.
+
+Another use-case for components is the storage of equivariant data, where a
+given irreducible representation might have multiple elements. For example, when
+handling spherical harmonics (which are the irreducible representation of the
+`group of 3D rotations`_ :math:`SO(3)`), all the spherical harmonics with the same
+:math:`\ell` and corresponding :math:`m` should be considered simultaneously:
+the different :math:`m` are **components** of a single irreducible
+representation.
+
+.. _group of 3D rotations: https://en.wikipedia.org/wiki/3D_rotation_group
+
+When handling the gradients of equivariant data, we quickly realize that we
+might need more than one component in a given :py:class:`TensorBlock`. Gradients
+with respect to positions of an equivariant representation based on spherical
+harmonics will need both the gradient direction :math:`x/y/z` and the spherical
+harmonics :math:`m` as components. This has two implications:
+
+- :py:class:`TensorBlock` can have an arbitrary number of components, which will
+  always occur *"in between"* samples and properties metadata;
+- when values in a :py:class:`TensorBlock` have components, and gradient with
+  respect to some parameter would add more components, the resulting gradient
+  components will contain first the new, gradient-specific components, and later
+  all of the components already present in the values.
```

Comment:
I would spin this into a new page for `What's in a Tensor Block?`, as these are as core of concepts like `TensorMap`, `Labels`, and `TensorBlock`, and keeping the page shorter will help with comprehension.